geoapps/inversion/line_sweep/driver.py

```diff
@@ -0,0 +1,110 @@
+#  (see LICENSE file at the root of this source code package).
+
+import json
+import os
+
+import numpy as np
+from geoh5py.groups import ContainerGroup, SimPEGGroup
+from geoh5py.ui_json import InputFile
+from geoh5py.workspace import Workspace
+from param_sweeps.driver import SweepDriver, SweepParams
+from param_sweeps.generate import generate
+
+from geoapps.inversion.driver import InversionDriver
+
+
+class LineSweepDriver(SweepDriver, InversionDriver):
+    def __init__(self, params):
+        # InversionDriver.__init__(self, params)
+        self.workspace = params.geoh5
+        self.pseudo3d_params = params
+        self.cleanup = params.cleanup
+        super().__init__(self.setup_params())
+
+    def run(self):
+        super().run()
+        with self.workspace.open(mode="r+"):
+            self.collect_results()
+        if self.cleanup:
+            self.file_cleanup()
+
+    def setup_params(self):
+        path = self.workspace.h5file.replace(".geoh5", ".json")
+        generate(
+            path, parameters=["line_id"], update_values={"conda_environment": "geoapps"}
+        )
+        ifile = InputFile.read_ui_json(
+            os.path.join(path.replace(".ui.json", "_sweep.ui.json"))
+        )
+        with self.workspace.open(mode="r"):
+            lines = self.pseudo3d_params.line_object.values
+        ifile.data["line_id_start"] = int(lines.min())
+        ifile.data["line_id_end"] = int(lines.max())
+        ifile.data["line_id_n"] = len(np.unique(lines))
+        sweep_params = SweepParams.from_input_file(ifile)
+        sweep_params.geoh5 = self.workspace
+        return sweep_params
+
+    def file_cleanup(self):
+        """Remove files associated with the parameter sweep."""
+        path = os.path.join(os.path.dirname(self.workspace.h5file))
+        with open(os.path.join(path, "lookup.json"), encoding="utf8") as f:
+            files = list(json.load(f))
+            for file in files:
+                os.remove(f"{os.path.join(path, file)}.ui.json")
+                os.remove(f"{os.path.join(path, file)}.ui.geoh5")
+
+        os.remove(os.path.join(path, "lookup.json"))
+        os.remove(os.path.join(path, "SimPEG.log"))
+        os.remove(os.path.join(path, "SimPEG.out"))
+        os.remove(
+            os.path.join(
+                path, self.workspace.h5file.replace(".ui.geoh5", "_sweep.ui.json")
+            )
+        )
+
+    @staticmethod
+    def line_files(path):
+        with open(os.path.join(path, "lookup.json"), encoding="utf8") as file:
+            line_files = {v["line_id"]: k for k, v in json.load(file).items()}
+        return line_files
+
+    def collect_results(self):
+        path = os.path.join(os.path.dirname(self.workspace.h5file))
+        files = LineSweepDriver.line_files(path)
+        lines = np.unique(self.pseudo3d_params.line_object.values)
+        results_group = SimPEGGroup.create(self.workspace, name="Pseudo3DInversion")
```

Comment:
Related to the why? above, technically the driver should already have its group.

Reply:
done